internal/corazawaf/rule.go

```diff
@@ -203,14 +228,31 @@ func (r *Rule) doEvaluate(phase types.RulePhase, tx *Transaction, cache map[tran
 	} else {
 		ecol := tx.ruleRemoveTargetByID[r.ID_]
 		for _, v := range r.variables {
-			// TODO(anuraaga): Support skipping variables based on phase for rule chains too.
-			if multiphaseEvaluation && !r.HasChain && r.ParentID_ == 0 {
+			if multiphaseEvaluation && r.ParentID_ == 0 && (!r.HasChain || phase >= r.chainMinPhase) {
 				min := minPhase(v.Variable)
 				// When multiphase evaluation is enabled, any variable is evaluated at its
-				// earliest possible phase, so we skip it if the rule refers to other phases
-				// to avoid double-evaluation.
-				if min != types.PhaseUnknown && min != phase {
-					continue
+				// earliest possible phase, so we make sure to skip in other phases.
+				if min != types.PhaseUnknown {
+					if !r.HasChain {
+						// For rules that have no chains, we know the variable is evaluated in its
+						// min phase and no other phase.
+						if min != phase {
+							continue
+						}
+					} else {
+						if min < r.chainMinPhase {
```

Comment:
This logic is sort of tricky but also intuitive enough. On the bright side, while iterating to get to it various FTW tests would fail so there seems to be some test coverage helping with these already.